From: gangmo_seong <gangmo_seong@tmax.co.kr>
Date: Mon, 25 Oct 2021 12:05:14 +0900
Subject: [bugfix] notification-center: disable function fix

- error fix early return in disable function
- separate subpanel logics to different function

[IMS] 272701
---
 extensions/notification-center/extension.js | 110 ++++++++++++++--------------
 1 file changed, 57 insertions(+), 53 deletions(-)

diff --git a/extensions/notification-center/extension.js b/extensions/notification-center/extension.js
index 1795d69..35f6aa2 100644
--- a/extensions/notification-center/extension.js
+++ b/extensions/notification-center/extension.js
@@ -40,7 +40,6 @@ function enable() {
 }
 
 function disable() {
-    log('disable notificationcenter');
     notificationCenter.undoChanges();
     notificationCenter.destroy();
 
@@ -942,68 +941,73 @@ const NotificationCenter = new LangClass({
 
         this.notificationSection._messages.forEach(message => this._setMessageLayout(message));
 
+        this.changeSubPanelUI();
+    },
+
+    changeSubPanelUI() {
         // apply UI changes also to sub panels
-        if (global.dashToPanel.panels) {
-            if (global.dashToPanel.panels.length < 2)
+        if (!global.dashToPanel)
+            return;
+        if (!global.dashToPanel.panels)
+            return;
+        if (global.dashToPanel.panels.length < 2)
+            return;
+
+        this.subDtActors = [];
+        this.subMessageLists = [];
+        this.subMessageListParents = [];
+        global.dashToPanel.panels.forEach(panel => {
+            if (panel === Main.panel.get_parent())
                 return;
-            this.subDtActors = [];
-            this.subMessageLists = [];
-            this.subMessageListParents = [];
-            global.dashToPanel.panels.forEach(panel => {
-                if (panel !== Main.panel.get_parent())
-                    this.changeSubPanelUI(panel);
-            });
-        } else {
-            // when dashtopanel loaded lately
-        }
-    },
 
-    changeSubPanelUI(panel) {
-        log('several panels');
-        log(global.dashToPanel.panels.length);
-        // remove dots and borders from sub panel datemenu
-        const dtActor = panel.statusArea.dateMenu.get_children()[0].get_children()[0];
-        this.subDtActors.push(dtActor);
-        panel.statusArea.dateMenu.get_children()[0].remove_actor(dtActor);
-        panel.statusArea.dateMenu.get_children()[0].remove_actor(panel.statusArea.dateMenu._indicator);
-
-        // remove messagelist from datemenu in sub panel
-        const messageList = panel.statusArea.dateMenu._messageList;
-        this.subMessageLists.push(messageList);
-        this.subMessageListParents.push(messageList.get_parent());
-        messageList.get_parent().remove_actor(messageList);
-
-        // use same default visitility value of main panel
-        panel.statusArea.dateMenu._weatherItem.visible = !this.prefs.get_boolean('hide-weather-section') && this.defaultWeatherItemVisibility;
-        panel.statusArea.dateMenu._clocksItem.visible =  !this.prefs.get_boolean('hide-clock-section') && this.defaultClocksItemVisibility;
-        panel.statusArea.dateMenu._date.visible =  !this.prefs.get_boolean('hide-date-section') && this.defaultDateVisibility;
+            // remove dots and borders from sub panel datemenu
+            const dtActor = panel.statusArea.dateMenu.get_children()[0].get_children()[0];
+            this.subDtActors.push(dtActor);
+            panel.statusArea.dateMenu.get_children()[0].remove_actor(dtActor);
+            panel.statusArea.dateMenu.get_children()[0].remove_actor(panel.statusArea.dateMenu._indicator);
+
+            // remove messagelist from datemenu in sub panel
+            const messageList = panel.statusArea.dateMenu._messageList;
+            this.subMessageLists.push(messageList);
+            this.subMessageListParents.push(messageList.get_parent());
+            messageList.get_parent().remove_actor(messageList);
+
+            // use same default visitility value of main panel
+            panel.statusArea.dateMenu._weatherItem.visible = !this.prefs.get_boolean('hide-weather-section') && this.defaultWeatherItemVisibility;
+            panel.statusArea.dateMenu._clocksItem.visible =  !this.prefs.get_boolean('hide-clock-section') && this.defaultClocksItemVisibility;
+            panel.statusArea.dateMenu._date.visible =  !this.prefs.get_boolean('hide-date-section') && this.defaultDateVisibility;
+        });
     },
 
-    revertSubPanelUI(panel) {
-    // revert dots and borders from sub panel datemenu
-        panel.statusArea.dateMenu.get_children()[0].insert_child_at_index(this.subDtActors.shift(), 0);
-        panel.statusArea.dateMenu.get_children()[0].add_actor(panel.statusArea.dateMenu._indicator);
+    revertSubPanelUI() {
+        // revert UI changes in sub panel
+        if (!global.dashToPanel)
+            return;
+        if (!global.dashToPanel.panels)
+            return;
+        if (global.dashToPanel.panels.length < 2)
+            return;
+
+        global.dashToPanel.panels.forEach(panel => {
+            if (panel === Main.panel.get_parent())
+                return;
+
+            // revert dots and borders from sub panel datemenu
+            panel.statusArea.dateMenu.get_children()[0].insert_child_at_index(this.subDtActors.shift(), 0);
+            panel.statusArea.dateMenu.get_children()[0].add_actor(panel.statusArea.dateMenu._indicator);
 
-        // revert messagelist of datemenu in sub panel
-        this.subMessageListParents.shift().insert_child_at_index(this.subMessageLists.shift(), 0);
+            // revert messagelist of datemenu in sub panel
+            this.subMessageListParents.shift().insert_child_at_index(this.subMessageLists.shift(), 0);
 
-        // revert visibility of items
-        panel.statusArea.dateMenu._weatherItem.visible = this.defaultWeatherItemVisibility;
-        panel.statusArea.dateMenu._clocksItem.visible = this.defaultClocksItemVisibility;
-        panel.statusArea.dateMenu._date.visible = this.defaultDateVisibility;
+            // revert visibility of items
+            panel.statusArea.dateMenu._weatherItem.visible = this.defaultWeatherItemVisibility;
+            panel.statusArea.dateMenu._clocksItem.visible = this.defaultClocksItemVisibility;
+            panel.statusArea.dateMenu._date.visible = this.defaultDateVisibility;
+        });
     },
 
     undoChanges() {
-
-        // revert UI changes in sub panel
-        if (global.dashToPanel.panels) {
-            if (global.dashToPanel.panels.length < 2)
-                return;
-            global.dashToPanel.panels.forEach(panel => {
-                if (panel !== Main.panel.get_parent())
-                    this.revertSubPanelUI(panel);
-            });
-        }
+        this.revertSubPanelUI();
 
         this.blinkIconStopIfBlinking();
 
